composer.json

```diff
@@ -40,11 +40,11 @@
     "require-dev": {
         "doctrine/coding-standard": "^8.1",
         "jetbrains/phpstorm-stubs": "^2019.1",
-        "phpstan/phpstan": "^0.12.40",
+        "phpstan/phpstan": "^0.12.54",
         "phpunit/phpunit": "^9.4",
-        "psalm/plugin-phpunit": "^0.10.0",
+        "psalm/plugin-phpunit": "^0.12.0",
         "symfony/console": "^2.0.5|^3.0|^4.0|^5.0",
-        "vimeo/psalm": "^3.17.2"
+        "vimeo/psalm": "^3.18.2"
```

Comment:
@morozov should we drop the `^` part here? that way we avoid discrepancies between local runs and CI (at the cost of a warning from Composer I think)

Reply:
The ^ doesn’t block later versions so why a warning?

Reply:
I checked and you are right, there is no warning. Maybe that changed, or I have faulty memory. That said, I was proposing to remove the `^`, that's when I thought you would get a warning, you won't get a warning with it that's for sure.

Reply:
> Maybe that changed, or I have faulty memory.

No, Composer would definitely trigger a warning in some cases if some dependency constraint was declared as a specific version.

> should we drop the `^` part here?

Why now? The specific version is locked in the lock file which we still commit to Git. I believe it should be done separately from this specific upgrade.

Reply:
I wasn't require we do it now, but in general. Now would be as good a time as any.

Reply:
> Now would be as good a time as any.

The way how we manage development dependencies is for the most part initiated by the work being done in #4386 which seems stuck. Once we decide to make a change, we'll make it for all dependencies at once.

This patch comes from an external contributor and just updates a dependency. I don't see much point in increasing the scope.